hcloud/data_source_hcloud_sshkeys.go

```diff
@@ -0,0 +1,66 @@
+package hcloud
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+func dataSourceHcloudSSHKeys() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHcloudSSHKeysRead,
+		Schema: map[string]*schema.Schema{
+			"with_selector": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"keys": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"fingerprint": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"public_key": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"labels": {
+							Type:     schema.TypeMap,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+func dataSourceHcloudSSHKeysRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*hcloud.Client)
+
+	labelSelector := d.Get("with_selector")
+	labelSelectorStr, _ := labelSelector.(string)
+
+	opts := hcloud.SSHKeyListOpts{
+		ListOpts: hcloud.ListOpts{
+			LabelSelector: labelSelectorStr,
+		},
+	}
+	keys, err := client.SSHKey.AllWithOpts(context.Background(), opts)
+	if err != nil {
+		return err
+	}
+	setSSHKeysSchema(d, keys)
```

Comment:
I would leave the method `setSSHKeysSchema` inline within this `datasource`  because we don't use it elsewhere.